vocabularies.md

```diff
@@ -0,0 +1,282 @@
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies/srsc"
+          }
+        }
+      },
+      {
+        "id": "common_types",
+        "name": "common_types",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies/common_types/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies/common_types"
+          }
+        }
+      },
+      {
+        "id": "common_iso_languages",
+        "name": "common_iso_languages",
+        "scrollable": true,
+        "hierarchical": false,
+        "type": "vocabulary",
+        "_links": {
+          "entries": {
+            "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies/common_iso_languages/entries"
+          },
+          "self": {
+            "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies/common_iso_languages"
+          }
+        }
+      }
+    ]
+  },
+  "_links": {
+    "self": {
+      "href": "https://dspace7.4science.cloud/server/api/integration/vocabularies"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 3,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+
+## Single Vocabulary
+**/api/integration/vocabularies/<:vocabulary-name>**
+
+Provide detailed information about a specific controlled vocabulary. The JSON response document is as follow
+```json
+{
+  "id": "srsc",
+  "name": "srsc",
+  "scrollable": false,
+  "hierarchical": true,
+  "preloadLevel": 1,
+  "type": "vocabulary"
+}
+```
+
+Attributes
+* id: the id of the vocabulary, it is the same than the name
+* name: see id
+* scrollable: if true mean that it is possible to scroll all the entries in the authority without providing a query parameter, see (vocabulary entries)[vocabularies.md#vocabulary-entries]
+* hierarchical: if true means that the vocabulary expose a tree structure where some entries are parent of others
+* preloadLevel: for hierarchical vocabularies express the preference to preload the tree at a specific level of depth (0 only the top nodes are shown, 1 also their childrens are preloaded and so on)
+
+Exposed links:
+* entries: get entries from the controlled vocabulary
+
+## Linked entities
+### controlled vocabulary entries
+**/api/integration/vocabularies/<:vocabulary-name>/entries**
+
+It returns the entries suggested by the vocabulary in response to the user query or a scrollable list, see below 
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* metadata: the metadata field for which the authority is used: mandatory
+* collection: the uuid of the collection where the item belong to: mandatory
```

Comment:
I have tried to remove the metadata and the collection from this endpoint completely but It requires a lot of update in the backend so at the end I would stay more close to the current implementation. The metadata is used by the SOLRAuthority to really filter the values see https://github.com/DSpace/DSpace/blob/30ad7676ef92df507cffb7a2040f5ef3ee5e56b9/dspace-api/src/main/java/org/dspace/content/authority/SolrAuthority.java#L79 the collection is also passed to all the authorities plugin moreover, it is required if we want to assure consistency about querying the right authority (oops vocabulary) for a given metadata and collection (thing about the case where one collection allow a free text dc.type when "all the other" stay with the traditional submission that use a value-pairs for it). `/api/integration/vocabularies/common_types/entries?metadata=dc.type&collection=<:uuid-special-collection>` would be not valid and should return 422 as noted in the response code

Reply:
@abollini : I completely understand why the `metadata` param would be needed, as it's clear that some vocabularies are only allowed for specific metadata fields or may return different results based on the metadata field.

I now also see your point that it *is* possible to configure (in submission-forms.xml) a single field (like dc.type) to act differently for different collections.  It might use different sets of value-pairs, or (as you noted) be free-text in one Collection and value-pairs in another.

This makes me wonder if maybe we need to move this entire endpoint to `/api/submission/vocabularies` (instead of `/api/integrations/vocabularies`).  That would clarify that this is *NOT* a generic "vocabularies" endpoint to be used anywhere in the system. It's really specific to the submission process & your submission configuration, and this is exactly why a `collection` param is required (as submission configs are specific to collections).   This is a tiny change here, but it clarifies in my mind why `collection` is mandatory, as it's mandatory for a submission.

We also may want to change the description of this `collection` param to clarify it's purpose.  So, we could say something like:
`collection : the uuid of the collection where the item is being submitted: mandatory.  The entries returned by a vocabulary may differ per collection based on your submission configuration.`